lockfile/lockfile.go

```diff
@@ -0,0 +1,84 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package lockfile
+
+import (
+	"errors"
+	sys "syscall" // XXX: if we are only targeting 15.04+, i.e. golang 1.5+, we should move to golang.org/x/sys
+)
+
+// ErrAlreadyLocked is returned when an attempts is made to lock an
+// already-locked FileLock.
+var ErrAlreadyLocked = errors.New("another snappy is running, try again later")
```

Comment:
I wonder if it makes sense to make this more generic (i.e. remove the "snappy" in the error). The ErrAlreadyLocked is already converted into a user-friendly (i18n) string in the frontend so it seems it won't harm the UI experience.